miniapps/dpg/convection-diffusion.cpp

```diff
@@ -0,0 +1,572 @@
+//              MFEM Ultraweak DPG example for convection-diffusion
+//
+// Compile with: make convection-diffusion
+//
+// sample runs
+// convection-diffusion -m ../../data/star.mesh -o 2 -ref 2 -theta 0.0 -eps 1e-1 -beta '2 3'
+// convection-diffusion -m ../../data/beam-hex.mesh -o 2 -ref 2 -theta 0.0 -eps 1e0 -beta '1 0 2'
+// convection-diffusion -m ../../data/inline-tri.mesh -o 3 -ref 2 -theta 0.0 -eps 1e-2 -beta '4 2' -sc
+
+// AMR runs
+// convection-diffusion  -o 3 -ref 5 -prob 1 -eps 1e-1 -theta 0.75
+// convection-diffusion  -o 2 -ref 9 -prob 1 -eps 1e-2 -theta 0.75
+// convection-diffusion  -o 3 -ref 9 -prob 1 -eps 1e-3 -theta 0.75 -sc
+
+// Description:
+// This example code demonstrates the use of MFEM to define and solve
+// the "ultraweak" (UW) DPG formulation for the convection-diffusion problem
+
+//     - εΔu + ∇⋅(βu) = f,   in Ω
+//                  u = u_0, on ∂Ω
+
+// It solves the following kinds of problems
+// (a) A manufactured solution where u_exact = sin(π * (x + y + z)).
+// (b) The 2D Erickson-Johnson problem
+
+// The DPG UW deals with the First Order System
+//     - ∇⋅σ + ∇⋅(βu) = f,   in Ω
+//        1/ε σ - ∇u  = 0,   in Ω
+//                  u = u_0, on ∂Ω
+
+// Ultraweak-DPG is obtained by integration by parts of both equations and the
+// introduction of trace unknowns on the mesh skeleton
+//
+// u ∈ L²(Ω), σ ∈ (L²(Ω))ᵈⁱᵐ
+// û ∈ H^1/2, σ̂ ∈ H^-1/2
+// -(βu , ∇v)  + (σ , ∇v)     + < f̂ ,  v  > = (f,v),   ∀ v ∈ H¹(Ω)
+//   (u , ∇⋅τ) + 1/ε (σ , τ)  + < û , τ⋅n > = 0,       ∀ τ ∈ H(div,Ω)
+//                                        û = u_0  on ∂Ω
+
+// Note:
+// f̂ := βu - σ
+// û := -u
+
+// -------------------------------------------------------------
+// |   |     u     |     σ     |   û       |     f̂    |  RHS    |
+// -------------------------------------------------------------
+// | v |-(βu , ∇v) | (σ , ∇v)  |           | < f̂ ,v > |  (f,v)  |
+// |   |           |           |           |          |         |
+// | τ | (u ,∇⋅τ)  | 1/ε(σ , τ)|  <û,τ⋅n>  |          |    0    |
+
+// where (v,τ) ∈  H¹(Ωₕ) × H(div,Ωₕ)
+
+#include "mfem.hpp"
+#include "util/weakform.hpp"
+#include "../common/mfem-common.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace std;
+using namespace mfem;
+using namespace mfem::common;
+
+enum prob_type
+{
+   manufactured,
+   EJ
```

Comment:
In `pconvection-diffusion.cpp`, there is a comment on `EJ` with a reference. That could be copied here.